xmtp/src/storage/encrypted_store/mod.rs

```diff
@@ -246,6 +246,36 @@ impl EncryptedMessageStore {
 
         Ok(msg_list)
     }
+
+    pub fn lock_refresh_job<F>(&self, kind: RefreshJobKind, cb: F) -> Result<(), StorageError>
```

Comment:
This feels right on the line of being too application specific to live in the store. Might move to somewhere else.